pkg/sidecar/sidecar.go

```diff
@@ -183,6 +232,132 @@ func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	return nil
 }
 
+func (s *Sidecar) writeJSON(fileName string, certs map[string]interface{}) {
+	file, err := json.Marshal(certs)
+	if err != nil {
+		s.config.Log.Errorf("Unable to parse certs: %v", err)
+	}
+
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	err = os.WriteFile(jsonPath, file, os.ModePerm)
+	if err != nil {
+		s.config.Log.Errorf("Unable to write JSON file: %v", err)
+	}
+}
+
+func (s *Sidecar) updateJWTBundle(jwkSet *jwtbundle.Set) {
+	s.config.Log.Info("Updating JWT bundle")
+
+	bundles := make(map[string]interface{})
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			s.config.Log.Errorf("Unable to marshal JWT bundle: %v", err)
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	s.writeJSON(s.config.JWTBundleFilename, bundles)
+}
+
+func (s *Sidecar) fetchJWTSVID(options ...workloadapi.ClientOption) (*jwtsvid.SVID, error) {
+	clientOptions := workloadapi.WithClientOptions(options...)
+
+	jwtSource, err := workloadapi.NewJWTSource(context.Background(), clientOptions)
+	if err != nil {
+		s.config.Log.Errorf("Unable to create JWTSource: %v", err)
+		return nil, err
+	}
+	defer jwtSource.Close()
+
+	jwtSVID, err := jwtSource.FetchJWTSVID(context.Background(), jwtsvid.Params{Audience: s.config.JWTAudience})
+	if err != nil {
+		s.config.Log.Errorf("Unable to fetch JWT SVID: %v", err)
+		return nil, err
+	}
+
+	_, err = jwtsvid.ParseAndValidate(jwtSVID.Marshal(), jwtSource, []string{s.config.JWTAudience})
+	if err != nil {
+		s.config.Log.Errorf("Unable to parse or validate token: %v", err)
+		return nil, err
+	}
+
+	return jwtSVID, nil
+}
+
+func createRetryIntervalFunc() func() time.Duration {
+	const (
+		initialBackoff = 1 * time.Second
+		maxBackoff     = 60 * time.Second
+		multiplier     = 2
+	)
+	backoffInterval := initialBackoff
+	return func() time.Duration {
+		currentBackoff := backoffInterval
+		// Update backoffInterval for next call, capped at maxBackoff
+		backoffInterval *= multiplier
+		if backoffInterval > maxBackoff {
+			backoffInterval = maxBackoff
+		}
+		return currentBackoff
+	}
+}
+
+func getRefreshInterval(svid *jwtsvid.SVID) time.Duration {
+	return time.Until(svid.Expiry)/2 + time.Second
+}
+
+func (s *Sidecar) performJWTSVIDUpdate(options ...workloadapi.ClientOption) (*jwtsvid.SVID, error) {
+	s.config.Log.Infof("Updating JWT SVID")
```

Comment:
Change this log level to `DEBUG` since we already have an `INFO` log line indicating the success of the update. Using `DEBUG` here would help in reducing log verbosity at the `INFO` level. Additionally, use the `Debug` method instead of `Debugf` since there isn't any string formatting required in this log statement.